src/FondOfSpryker/Glue/ProductListsRestApi/Processor/ProductLists/ProductListsReader.php

```diff
@@ -0,0 +1,244 @@
+    }
+
+    /**
+     * @param \Spryker\Glue\GlueApplication\Rest\Request\Data\RestRequestInterface $restRequest
+     *
+     * @return \Spryker\Glue\GlueApplication\Rest\JsonApi\RestResponseInterface
+     */
+    public function findAllProductListsByCustomerId(RestRequestInterface $restRequest): RestResponseInterface
+    {
+        $restResponse = $this->restResourceBuilder->createRestResponse();
+
+        $customerTransfer = (new CustomerTransfer())
+            ->setIdCustomer($restRequest->getRestUser()->getSurrogateIdentifier());
+
+        $productListCollectionTransfer = $this->productListCustomerClient
+            ->getProductListCollectionByCustomerId($customerTransfer);
+
+        if ($productListCollectionTransfer->getProductLists()->count() === 0) {
+            return $this->restApiError->addProductListNotFoundError($restResponse);
+        }
+
+        return $this->addProductListCollectionTransferToResponse($productListCollectionTransfer, $restResponse);
+    }
+
+    /**
+     * @param \Spryker\Glue\GlueApplication\Rest\Request\Data\RestRequestInterface $restRequest
+     *
+     * @return \Spryker\Glue\GlueApplication\Rest\JsonApi\RestResponseInterface
+     */
+    public function findProductListByUuid(RestRequestInterface $restRequest): RestResponseInterface
+    {
+        $restResponse = $this->restResourceBuilder->createRestResponse();
+
+        if (!$restRequest->getResource()->getId()) {
+            return $this->restApiError->addProductListUuidMissingError($restResponse);
+        }
+
+        $productListTransfer = (new ProductListTransfer())
+            ->setUuid($restRequest->getResource()->getId());
+
+        $productListResponseTransfer = $this->productListsClient
+            ->findProductListByUuid($productListTransfer);
+
+        if (!$productListResponseTransfer->getIsSuccessful()) {
+            return $this->restApiError->addProductListNotFoundError($restResponse);
+        }
+
+        if (!$this->productListIsFromCustomerOrCompanyUser($productListResponseTransfer->getProductList(), $restRequest->getRestUser())) {
+            return $this->restApiError->addProductListNoPermissionError($restResponse);
+        }
+
+        $restProductListsAttributesTransfer = $this->productListsMapper
+            ->mapProductListTransferToRestProductListResponseAttributesTransfer(
+                $productListResponseTransfer->getProductList(),
+                new RestProductListsAttributesTransfer()
+            );
+
+        return $this->addRestProductListsAttributesTransferToResponse($restProductListsAttributesTransfer, $restResponse);
+    }
+
+    /**
+     * @param \Generated\Shared\Transfer\ProductListTransfer $productListTransfer
+     * @param \Generated\Shared\Transfer\RestUserTransfer|null $restUserTransfer
+     *
+     * @return bool
+     */
+    protected function productListIsFromCustomerOrCompanyUser(ProductListTransfer $productListTransfer, ?RestUserTransfer $restUserTransfer): bool
```

Comment:
you should split the parameters in more than on line it exceds the maxmium length limit

protected function productListIsFromCustomerOrCompanyUser(
ProductListTransfer $productListTransfer, ?
RestUserTransfer $restUserTransfer
): bool {